src/FastSerialization/StreamReaderWriter.cs

```diff
@@ -368,11 +372,12 @@ public void Dispose()
             // Make sure we don't exceed max possible size
             if (bytes.Length < (maxLength / 3) * 2)
                 newLength = (bytes.Length / 2) * 3;
-            else if (bytes.Length < maxLength)
-                newLength = maxLength;
+            else if (bytes.Length < maxLength - sizeof(long))      // Write(long) expects Makespace to make at 8 bytes of space.   
```

Comment:
:bulb: I always find the comments more readable when they are on the preceding line (or immediately inside the block when commenting on a condition) rather than at the end, but either way works